website/content/docs/upgrading/upgrade-to-1.10.x.mdx

```diff
@@ -52,3 +52,26 @@ were 26 characters; they now have a minimum of 95 characters. However, existing
 still work.
 
 Refer to the [Service Side Consistent Token FAQ](/docs/faq/ssct) for details.
+
+## OIDC Provider Built-in Resources
+
+In Vault 1.9, the [OIDC identity provider](/docs/secrets/identity/oidc-provider) feature
+was released as a tech preview. In Vault 1.10, built-in resources were introduced to the
+OIDC provider system to reduce configuration steps and enhance usability.
+
+The following built-in resources are included in each Vault namespace starting with Vault
+1.10:
+
+ - A "default" OIDC provider that's usable by all client applications
+ - A "default" key for signing and verification of ID tokens
+ - An "allow_all" assignment which authorizes all Vault entities to authenticate via a
+   client application
+
+If you created an [OIDC provider](/api-docs/secret/identity/oidc-provider#create-or-update-a-provider)
+with the name "default", [key](/api-docs/secret/identity/tokens#create-a-named-key) with the
+name "default", or [assignment](/api-docs/secret/identity/oidc-provider#create-or-update-an-assignment)
+with the name "allow_all" using the Vault 1.9 tech preview, the installation of these built-in
+resources will be skipped without an error. We recommend deleting any resources that have
```

Comment:
Interesting, I didn't know that we simply skipped these builtin resources. Do we make any assumptions in the code on these literal values (e.g. "allow_all")?

Reply:
Ya, we check if they exist, and if so, skip creation. See https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc_provider.go#L2386. 

There is an assumption made based on the literal value at https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc_provider.go#L2316. This makes me wonder if we should more strongly emphasize the need to delete preexisting resources that collide with these names.

Reply:
Yeah, I have similar thoughts too... Another thing to consider is that we disable modification of these entries after the upgrade so it'd be really hard for them to delete these entries afterwards unless they resort to using `sys/raw/`.

Reply:
I've added a stronger emphasis on this in 55e14cf. I agree. There are already checks to ensure that deletion/modification don't happen in the 1.10 code. For example, https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc_provider.go#L660-L663 and https://github.com/hashicorp/vault/blob/main/vault/identity_store_oidc_provider.go#L760-L763.